src/applications/virtual-agent/components/webchat/helpers/startConvoAndTrackUtterancesHelpers.js

```diff
@@ -121,3 +122,18 @@ export const processIncomingActivity = ({ action, dispatch }) => () => {
     sendWindowEvent('rxSkill');
   }
 };
+
+export const processMicrophoneActivity = ({ action }) => () => {
+  const isRxSkill = sessionStorage.getItem(IS_RX_SKILL);
+  if (action.payload.dictateState === 3) {
+    recordEvent({
+      event: 'chatbot-microphone-enable',
+      topic: isRxSkill ? 'prescriptions' : undefined,
+    });
+  } else if (action.payload.dictateState === 0) {
+    recordEvent({
+      event: 'chatbot-microphone-disable',
+      topic: isRxSkill ? 'prescriptions' : undefined,
+    });
+  }
+};
```

Comment:
should this go in tracking.js with the others?

Reply:
I had put processMicrophoneActivity in startConvoAndTrackUtterancesHelpers because it's called from startConvoAndTrackUtterances where as methods in the tracking file were are called from WebChat.jsx and worked differently. The code isn't structured very well and should be refactored (files in mind: startConvoAndTrackUtterances, startConvoAndTrackUtterancesHelpers, webchat.jsx, tracking.js) but I feel like moving processMicrophoneActivity would just add to the confusion.